docs/swift/interceptors.md

```diff
@@ -6,40 +6,85 @@ sidebar_position: 5
 Interceptors are a powerful way to observe and mutate outbound and inbound
 headers, data, trailers, and errors both for unary APIs and streams.
 
-An interceptor is instantiated **once for each request** and provides a set of
-closures that are invoked by the client during the lifecycle of that request.
-Each closure provides the ability for the interceptor to observe and store
-state, as well as the option to mutate the outbound or inbound content.
+Each interceptor is instantiated **once per request or stream** and
+provides a set of closures that are invoked by the client during the lifecycle
+of that call. Each closure allows the interceptor to observe and store
+state, as well as to mutate outbound or inbound content.
+
+Every interceptor has the opportunity to perform asynchronous work before passing a potentially
+altered value to the next interceptor in the chain. When the end of the chain is reached, the
+final value is passed to the networking client where it is sent to the server (outbound request)
+or to the caller (inbound response).
+
+Interceptors may also fail outbound requests before they're sent, thus preventing subsequent
+interceptors from being invoked and returning a specified error back to the original caller.
```

Comment:
```suggestion
Interceptors may also fail outbound requests before they are sent; subsequent 
interceptors in the chain will not be invoked, and the error will be returned to the original caller.
```

I think removing the 'thus preventing' language better communicates that subsequent interceptors not being invoked is more just a side effect of the failure rather than the key purpose of the failure (which is, as stated, to prevent the *request* from being sent).